packages/authentication/cardstack/middleware.js

```diff
@@ -163,9 +164,28 @@ class Authentication {
     throw new Error(`No such authentication source "${name}"`, { status: 404 });
   }
 
+  async _sessionMeta(ctxt) {
+    let config, ip;
+
+    try {
+       config = await this.searcher.getFromControllingBranch(Session.INTERNAL_PRIVILEGED, 'plugin-configs', '@cardstack/authentication');
+    } catch (e) {
+      // no config for the authentication plugin, assume header should not be respected
+    }
+
+    if (config && config.data.attributes['plugin-config'] && config.data.attributes['plugin-config']['allow-x-forwarded-for']) {
```

Comment:
Instead of the field name `plugin-config`, we can use the existing `params` field which serves exactly the same purpose in multiple other places (a bag of plugin-specific configuration that is untyped from the perspective of the JSON:API schema).